org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextValidator.java

```diff
@@ -258,6 +258,13 @@ else if (GrammarUtil.isDatatypeRule((ParserRule) firstRule)) {
 					firstRule,
 					XtextPackage.Literals.ABSTRACT_RULE__NAME, 
 					ValidationMessageAcceptor.INSIGNIFICANT_INDEX);
+		} 
+		else if (((ParserRule) firstRule).isFragment()) {
```

Comment:
nitpick: we usually place the `else` after the closing braces of the previous block

Reply:
Me too, I wanted to imitate the rest of the method.
Would you prefer it on the same line?

Reply:
This method is unfortunately an example where it is handled differently than in the rest of the file. So I'd say please change the one above also to match the file's common formatting.